homeassistant/components/ring/binary_sensor.py

```diff
@@ -33,49 +32,60 @@
                 continue
 
             for device in devices[device_type]:
-                sensors.append(RingBinarySensor(ring, device, sensor_type))
+                sensors.append(
+                    RingBinarySensor(config_entry.entry_id, ring, device, sensor_type)
+                )
 
-    async_add_entities(sensors, True)
+    async_add_entities(sensors)
 
 
-class RingBinarySensor(BinarySensorDevice):
+class RingBinarySensor(RingEntityMixin, BinarySensorDevice):
     """A binary sensor implementation for Ring device."""
 
-    def __init__(self, ring, device, sensor_type):
+    _active_alert = None
+
+    def __init__(self, config_entry_id, ring, device, sensor_type):
         """Initialize a sensor for Ring device."""
-        self._sensor_type = sensor_type
+        super().__init__(config_entry_id, device)
         self._ring = ring
-        self._device = device
+        self._sensor_type = sensor_type
         self._name = "{0} {1}".format(
-            self._device.name, SENSOR_TYPES.get(self._sensor_type)[0]
+            self._device.name, SENSOR_TYPES.get(sensor_type)[0]
         )
-        self._device_class = SENSOR_TYPES.get(self._sensor_type)[2]
+        self._device_class = SENSOR_TYPES.get(sensor_type)[2]
         self._state = None
-        self._unique_id = f"{self._device.id}-{self._sensor_type}"
-        self._disp_disconnect = None
+        self._unique_id = f"{device.id}-{sensor_type}"
+        self._update_alert()
 
     async def async_added_to_hass(self):
         """Register callbacks."""
-        self._disp_disconnect = async_dispatcher_connect(
-            self.hass, SIGNAL_UPDATE_RING, self._update_callback
-        )
+        await super().async_added_to_hass()
+        self.ring_objects["dings_data"].add_listener(self._dings_update_callback)
+        self._dings_update_callback()
 
     async def async_will_remove_from_hass(self):
         """Disconnect callbacks."""
-        if self._disp_disconnect:
-            self._disp_disconnect()
-            self._disp_disconnect = None
+        await super().async_will_remove_from_hass()
+        self.ring_objects["dings_data"].remove_listener(self._dings_update_callback)
 
     @callback
-    def _update_callback(self):
+    def _dings_update_callback(self):
         """Call update method."""
-        self.async_schedule_update_ha_state(True)
-        _LOGGER.debug("Updating Ring binary sensor %s (callback)", self.name)
+        self._update_alert()
+        self.async_write_ha_state()
 
-    @property
-    def should_poll(self):
-        """Return False, updates are controlled via the hub."""
-        return False
+    @callback
+    def _update_alert(self):
+        """Update active alert."""
+        self._active_alert = next(
+            (
+                alert
+                for alert in self._ring.active_alerts()
```

Comment:
Is this just accessing a local data cache?

Reply:
Yes